app/serializers/preprint.js

```diff
@@ -1,4 +1,19 @@
 import OsfSerializer from 'ember-osf/serializers/osf-serializer';
 
 export default OsfSerializer.extend({
+    serialize(snapshot) {
+        // Normal OSF serializer strips out relationships. We need to add back primaryFile for this endpoint
+        let res = this._super(...arguments);
+        res.data.relationships = {
+            // Not sure what the name of this key comes from, but it's required.
```

Comment:
This can actually be called `primary_file` to match the relationship in the preprint serializer and it'll work just fine.  The relationship getting properly updated actually depends on getting the `type` field right - in this case `primary_file`. We're looking into if that's just the way our API currently handles relationships, or if we can change it.